include/picongpu/fields/laserProfiles/PulseFromSpectrum.hpp

```diff
@@ -0,0 +1,350 @@
+                    static constexpr float_X PULSE_LENGTH
+                        = float_X(Params::PULSE_LENGTH_SI / UNIT_TIME); // unit: seconds (1 sigma)
+                    static constexpr float_X AMPLITUDE
+                        = float_X(Params::AMPLITUDE_SI / UNIT_EFIELD); // unit: volt / meter
+                    static constexpr float_X PULSE_INIT = float_X(Params::PULSE_INIT); // unit: none
+                    static constexpr float_X GDD
+                        = float_X(Params::GDD_SI / (UNIT_TIME * UNIT_TIME)); // unit: seconds^2
+                    static constexpr float_X TOD
+                        = float_X(Params::TOD_SI / (UNIT_TIME * UNIT_TIME * UNIT_TIME)); // unit: seconds^3
+                    static constexpr float_X W0 = float_X(Params::W0_SI / UNIT_LENGTH); // unit: meter
+                    static constexpr float_X FOCUS_POS = float_X(Params::FOCUS_POS_SI / UNIT_LENGTH); // unit: meter
+                    static constexpr float_X INIT_TIME = float_X(
+                        (Params::PULSE_INIT * Params::PULSE_LENGTH_SI)
+                        / UNIT_TIME); // unit: seconds (full inizialisation length)
+
+                    /* Rayleigh-Length in y-direction
+                     */
+                    static constexpr float_X Y_R = float_X(PI * W0 * W0 / WAVE_LENGTH); // unit: meter
+                };
+            } // namespace pulseFromSpectrum
+
+            namespace acc
+            {
+                template<typename T_Unitless>
+                struct PulseFromSpectrum : public T_Unitless
+                {
+                    using Unitless = T_Unitless;
+
+                    float3_X m_elong;
+                    float_X m_phase;
+                    float_X m_currentStep;
+                    typename FieldE::DataBoxType m_dataBoxE;
+                    DataSpace<simDim> m_offsetToTotalDomain;
+                    DataSpace<simDim> m_superCellToLocalOriginCellOffset;
+
+                    /** waist of the laser beam
+                     * @param y_to_focus y-coordinate relative to focus-position
+                     */
+                    HDINLINE float_X Waist(float_X y_to_focus)
+                    {
+                        return Unitless::W0 * math::sqrt(1.0 + math::pow(y_to_focus / Unitless::Y_R, 2.0));
+                    }
+
+                    /** inverse radius of the beam-curvature
+                     * @param y_to_focus y-coordinate relative to focus-position
+                     */
+                    HDINLINE float_X R_inv(float_X y_to_focus)
+                    {
+                        //??? if Unitless::Y_R is inserted in (97) in pow() --> compile error???, why not in line 86,
+                        // why does this workaround work?
+                        float_X y_R = Unitless::Y_R;
+                        return y_to_focus / (math::pow(y_to_focus, 2.0) + math::pow(y_R, 2.0));
```

Comment:
Values passed to `pow` must be lvalues. `Unitless::Y_R` is constexpr and therefore does not have any storage. This workaround is valid and IMO does not require a docstring.

Reply:
Why do we require it in our math functions at all? Probably needs to be changed to pass-by-value (outside of this PR)?